### ISSUE TITLE
Wrong UTC logic in DateTime

### ISSUE THREAD

**Issue by hardhub** [bridgedotnet/Bridge#3770]
It is critical for my project since it is based on heavy date-time calculation.

### Steps To Reproduce

https://dev.deck.net/0682aea53a9ef3fc52165566fd051fff
https://dotnetfiddle.net/2meO3I

```csharp
public class Program
{
    public static void Main()
    {
        var time = DateTime.Parse("2018-11-04T00:00:00.000Z");
        
        time = DateTime.SpecifyKind(time, DateTimeKind.Utc);
        
        Console.WriteLine(time.Kind.ToString());
            
        Console.WriteLine(time.ToString("u"));
    }
}
```

### Expected Result

```
Utc
2018-11-04 00:00:00Z
```

### Actual Result

```
Utc
2018-11-04 03:00:00Z
```

**Comment by hardhub** [comment]
It is also related to https://github.com/bridgedotnet/Bridge/issues/3771

If it is UTC+2 it will be 2018-11-04 **02**:00:00Z and so on

**Comment by geoffreymcgill** [comment]
Thanks for the report. This will take some time to investigate.

**Comment by geoffreymcgill** [comment]
I believe we have this issue fixed. You can test your original Deck.NET sample in our **dev** instance:

https://dev.deck.net/0682aea53a9ef3fc52165566fd051fff

We still have a lot more testing to do, but if everything goes well, this fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by hardhub** [comment]
Yes, it looks so.  Probably that is already fixed when DateTome logic is updated for other issues.

**Comment by geoffreymcgill** [comment]
There were several defects fixed within the DateTime js logic specific to this issue and other open issues. These fixes will be sent as a pull-request once further testing is complete.

**Comment by geoffreymcgill** [comment]
This scenario has been fixed. After another review and final testing, the fix will be included in the upcoming Bridge 17.8.0 release.

You can confirm with the following dev.deck.net sample:

https://dev.deck.net/0682aea53a9ef3fc52165566fd051fff

**Comment by fabriciomurta** [comment]
A full test over all time zones have been undegone and all looks good, so the fix for this will be available next Bridge release!
